index.php

```diff
@@ -28,260 +29,330 @@
 
-chess_get_games();
+    global $xoopsDB, $xoopsTpl;
 
-include_once XOOPS_ROOT_PATH . '/footer.php';
-/**#@-*/
+    // ----------
 
-/**
- * Generate lists of games and challenges.
- */
-function chess_get_games()
-{
-	global $xoopsDB, $xoopsTpl;
-
-// ----------
-// user input
-// ----------
-
-	// offset of first row of challenges table to display (default to 0)
-	$cstart = intval(isset($_POST['cstart']) ? $_POST['cstart'] : @$_GET['cstart']);
-	// offset of first row of games table to display (default to 0)
-	$gstart = intval(isset($_POST['gstart']) ? $_POST['gstart'] : @$_GET['gstart']);
-	// challenges display option
-	$cshow  = intval(isset($_POST['cshow'])  ? $_POST['cshow']  : @$_GET['cshow']);
-	// games display option 1
-	$gshow1 = intval(isset($_POST['gshow1']) ? $_POST['gshow1'] : @$_GET['gshow1']);
-	// games display option 2
-	$gshow2 = intval(isset($_POST['gshow2']) ? $_POST['gshow2'] : @$_GET['gshow2']);
-
-	// set show-options to default if undefined
-	if (!$cshow) {
-		$cshow = _CHESS_SHOW_CHALLENGES_BOTH;
-	}
-	if (!$gshow1) {
-		$gshow1 = _CHESS_SHOW_GAMES_BOTH;
-	}
-	if (!$gshow2) {
-		$gshow2 = _CHESS_SHOW_GAMES_UNRATED;
-	}
-
-	// get maximum number of items to display on a page, and constrain it to a reasonable value
-	$max_items_to_display = chess_moduleConfig('max_items');
-	$max_items_to_display = min(max($max_items_to_display, 1), 1000);
-
-	$xoopsTpl->assign('chess_date_format', _MEDIUMDATESTRING);
-
-	// user IDs that will require mapping to usernames
-	$userids = array();
-
-// -----
-// games
-// -----
-
-	// Two queries are performed, one without a limit clause to count the total number of rows for the page navigator,
-	// and one with a limit clause to get the data for display on the current page.
-	// SQL_CALC_FOUND_ROWS and FOUND_ROWS(), available in MySQL 4.0.0, provide a more efficient way of doing this.
-
-	$games_table = $xoopsDB->prefix('chess_games');
-
-	$where = 'white_uid != black_uid';
-	switch($gshow1) {
-		case 1:
-			$where .= " AND pgn_result = '*'";
-			break;
-		case 2:
-			$where .= " AND pgn_result != '*'";
-			break;
-	}
-	if ($gshow2 == 1) {
-		$where .= " AND is_rated = '1'";
-	}
-
-	$result = $xoopsDB->query("SELECT COUNT(*) FROM $games_table WHERE $where");
-	list($num_games) = $xoopsDB->fetchRow($result);
-	$xoopsDB->freeRecordSet($result);
-
-	$result = $xoopsDB->query(trim("
+    // user input
+
+    // ----------
+
+    // offset of first row of challenges table to display (default to 0)
+
+    $cstart = (int)($_POST['cstart'] ?? @$_GET['cstart']);
+
+    // offset of first row of games table to display (default to 0)
+
+    $gstart = (int)($_POST['gstart'] ?? @$_GET['gstart']);
+
+    // challenges display option
+
+    $cshow = (int)($_POST['cshow'] ?? @$_GET['cshow']);
+
+    // games display option 1
+
+    $gshow1 = (int)($_POST['gshow1'] ?? @$_GET['gshow1']);
+
+    // games display option 2
+
+    $gshow2 = (int)($_POST['gshow2'] ?? @$_GET['gshow2']);
```

Comment:
Recommend use of \Xmf\Request to process incoming vars - this was in my 1.08 Alpha 1 commit
```suggestion
// offset of first row of challenges table to display (default to 0)
$cstart = Request::getInt('cstart', 0);

// offset of first row of games table to display (default to 0)
$gstart = Request::getInt('gstart', 0);

// challenges display option
$cshow = Request::getInt('cshow', 0);

// games display option 1
$gshow1 = Request::getInt('gshow1', 0);

// games display option 2
$gshow2 = Request::getInt('gshow2', 0);
```